### PR TITLE
Website access management

### LINKED ISSUE #85: Manage access as previously granted
Goal:

The user should be able to check access on a per site basis and either approve/reject these.

Actions:

- storage permissions to storage
- page where the user to reject or approve based on sites that requested access

Similar to #10

### PR DESCRIPTION
**Pull request by Tbaut**
closes #85

todo:
- [x] adjust menu icon
- [x] verify the look and scrolling with many items

 
manage-access

### CODE REVIEW

**Comment by Tbaut** [comment]
A long list should be fine now. Here's a demo with filtering.
scroll

**Comment by jacogr** [comment]
where does that long subsocial list come from?

**Comment by Tbaut** [comment]
> where does that long subsocial list come from?
Me inserting it manually for testing :)

File: packages/extension-ui/src/Popup/AuthManagement/WebsiteEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+function WebsiteEntry ({ className = '', info, toggleAuth, url }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+
+  const switchAccess = useCallback(() => {
+    toggleAuth(url);
+  }, [toggleAuth, url]);
+
+  return (
+
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/AuthManagement/WebsiteEntry.tsx:29]:
```suggestion
```

File: packages/extension-ui/src/Popup/AuthManagement/WebsiteEntry.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <div className={`${className} ${info.isAllowed ? 'allowed' : 'denied'}`}>
+      <div className={'url'}>
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/AuthManagement/WebsiteEntry.tsx:31]:
```suggestion
      <div className='url'>
```

File: packages/extension-ui/src/Popup/AuthManagement/index.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+function AuthManagement ({ className }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const [authList, setAuthList] = useState<AuthUrls | null>(null);
+  const [filter, setFilter] = useState('');
+
+  useEffect(() => {
+    getAuthList().then(
+      ({ list }) => setAuthList(list)
+    ).catch((e) => console.error(e)
+    );
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/AuthManagement/index.tsx:30]:
```suggestion
    getAuthList()
      .then(({ list }) => setAuthList(list))
      .catch((e) => console.error(e));
    );
```

File: packages/extension-ui/src/Popup/AuthManagement/index.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  }, []);
+
+  const _onChangeFilter = useCallback((filter: string) => {
+    setFilter(filter);
+  }, []);
+
+  const toggleAuth = useCallback((url: string) => {
+    toggleAuthorization(url).then(({ list }) => {
+      setAuthList(list);
+    }).catch(console.error);
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/AuthManagement/index.tsx:40]:
```suggestion
    toggleAuthorization(url)
      .then(({ list }) => setAuthList(list))
      .catch(console.error);
```

File: packages/extension-ui/src/Popup/AuthManagement/index.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    <>
+      <Header
+        showBackArrow
+        smallMargin
+        text={t<string>('Manage Website Access')}
+      />
+      <>
+        <InputFilter
+          onChange={_onChangeFilter}
+          placeholder={t<string>('website.com')}
```
Comment (jacogr) [inline packages/extension-ui/src/Popup/AuthManagement/index.tsx:53]:
```suggestion
          placeholder={t<string>('example.com')}
```

website.com existis, example is used for examples :) ... one of those where somebody somewhere will complain.

File: packages/extension-ui/src/components/InputFilter.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      autoCapitalize='off'
+      autoCorrect='off'
+      autoFocus
+      onChange={onChangeFilter}
+      placeholder={placeholder}
+      spellCheck={false}
+      type='text'
+      value={value}
+    />
+  </div>;
```
Comment (jacogr) [inline packages/extension-ui/src/components/InputFilter.tsx:32]: Prefer this in a `(...)`

**Review by jacogr — COMMENTED** [review]
Just some small code-related comments, but nothing functional, even as-is it is great.

File: packages/extension-ui/src/components/InputFilter.tsx (hunk 6 above)
Comment (Tbaut) [reply packages/extension-ui/src/components/InputFilter.tsx]: so do I, usually. Changed it.

**Review by jacogr — APPROVED** [review]
Excellent. One of the most-requested features lately.

**Pull request merged**